### PR TITLE
drivers: clock_control: stm32u5 enables the VCO input and EPOD

### LINKED ISSUE #41606: stm32u5: Re-implement VCO input and EPOD configuration
**Describe the bug**
During implementation of https://github.com/zephyrproject-rtos/zephyr/pull/41310, VCO input and EPOD configuration were removed.
Initial tests didn't show any adverse effect, but it would be saner to re-instantiate this piece of code.

Instead of doing it directly in #41310, it is proposed to do it on top of https://github.com/zephyrproject-rtos/zephyr/pull/41585,
taking benefit of tests and code optimization provided.

### PR DESCRIPTION
**Pull request by FRASTM**
With the stm32U5, when the PLL1 source is MSI,
the VCO input and EPOD configuration must be configured
before the PLL1 is enabled. The config to write depends on the freq.

Following the https://github.com/zephyrproject-rtos/zephyr/pull/41310
The VCO input and EPOD configuration are restored for the stm32U5x

Fixes https://github.com/zephyrproject-rtos/zephyr/issues/41606

Signed-off-by: Francois Ramu <francois.ramu@st.com>

### CODE REVIEW

**Comment by FRASTM** [comment]
in the config_src_sysclk_pll function, when the PLL source is **not MSI** then STM32_MSIS_RANGE is not defined but  the code portion
```
 if (IS_ENABLED(STM32_PLL_SRC_MSIS)) {
... 
```
exists though  --> build error

**Comment by FRASTM** [comment]
Rebase on 4a702ea21ef82b301464551807358f8490951e45
Implement the sequence from the refMan for dynamic voltage scaling management : 
- 1 Add a function `set_epod_booster()`  to configure the RCC_PLL1CFGR PLL1MBoost which generate the epod clock freq between 4 and 16MHz (prescaler)  
- 2 3.The VOS is programmed by the `set_regu_voltage()` function  
- 4  set BOOSTEN in the PWR_VOSR and wait for flag BOOSTRDY

**Review by ABOSTM — CHANGES_REQUESTED** [review]

File: drivers/clock_control/clock_stm32_ll_u5.c [hunk 1]
```diff
@@ -454,8 +484,14 @@ static int set_up_plls(void)
 
 	LL_RCC_PLL1_SetDivider(STM32_PLL_M_DIVISOR);
 
+	/* Set VCO Input before enabling the PLL, depends on freq used for PLL1 */
 	LL_RCC_PLL1_SetVCOInputRange(vco_input_range);
 
+	/* Enable EPOD booster and wait for booster ready flag set */
+	LL_PWR_EnableEPODBooster();
```
[the remark this answers is not in the document]
Comment (FRASTM) [reply drivers/clock_control/clock_stm32_ll_u5.c:491]: The EPOD prescaler has a (reset) default value which is correct when the freq does not require a divider.

File: drivers/clock_control/clock_stm32_ll_u5.c [hunk 2]
```diff
@@ -438,6 +466,8 @@ static int set_up_plls(void)
 		/* Main PLL configuration and activation */
 		LL_RCC_PLL1_SetMainSource(LL_RCC_PLL1SOURCE_HSE);
 	} else if (IS_ENABLED(STM32_PLL_SRC_MSIS)) {
+		/* Configure the EPOD booster before increasing the system clock freq */
+		set_epod_booster();
```
[the remark this answers is not in the document]
Comment (FRASTM) [reply drivers/clock_control/clock_stm32_ll_u5.c:470]: Only when the PLL source is MSI, the epod freq can require a divider to "to generate a booster
clock frequency between 4 and 16 MHz"

**Comment by FRASTM** [comment]
Adding more comment to explain the configuration
Possibly adding a reset epod prescaler in case it was set > 0 earlier and then reduced (by low power mode for example) then

**Comment by FRASTM** [comment]
Rebase on 9a9fc2b8a5ddf81e44330d9cbebaeba360651e68
Add more details about the config of the set_epod_booster() 
Reset the EPOD prescaler before updating.

**Pull request closed without merging**